primitives/beefy/src/witness.rs

```diff
@@ -110,37 +128,119 @@ mod tests {
 		(sig1.into(), sig2.into())
 	}
 
-	fn signed_commitment() -> TestSignedCommitment {
+	///generates mock aggregatable bls signature for generating test commitment
+	///BLS signatures
+	fn mock_bls_signatures() -> (BLSSignature, BLSSignature) {
+		let store: SyncCryptoStorePtr = KeyStore::new().into();
+
+		let mut alice = sp_core::bls::Pair::from_string("//Alice", None).unwrap();
+		let _ =
+			SyncCryptoStore::insert_unknown(&*store, KEY_TYPE, "//Alice", alice.public().as_ref())
+				.unwrap();
+
+		let msg = b"This is the first message";
+		let sig1 = alice.sign(msg);
+
+		let msg = b"This is the second message";
+		let sig2 = alice.sign(msg);
+
+		(sig1.into(), sig2.into())
+	}
+
+	fn ecdsa_signed_commitment() -> TestSignedCommitment {
 		let payload = Payload::from_single_entry(
 			known_payloads::MMR_ROOT_ID,
 			"Hello World!".as_bytes().to_vec(),
 		);
 		let commitment: TestCommitment =
 			Commitment { payload, block_number: 5, validator_set_id: 0 };
 
-		let sigs = mock_signatures();
+		let sigs = mock_ecdsa_signatures();
 
 		SignedCommitment { commitment, signatures: vec![None, None, Some(sigs.0), Some(sigs.1)] }
 	}
 
+	fn ecdsa_and_bls_signed_commitment() -> TestBLSSignedCommitment {
+		let payload =
+			Payload::new(known_payload_ids::MMR_ROOT_ID, "Hello World!".as_bytes().to_vec());
+		let commitment: TestCommitment =
+			Commitment { payload, block_number: 5, validator_set_id: 0 };
+
+		let ecdsa_sigs = mock_ecdsa_signatures();
+		let bls_sigs = mock_bls_signatures();
+
+		SignedCommitment {
+			commitment,
+			signatures: vec![
+				None,
+				None,
+				Some(ECDSABLSSignaturePair(ecdsa_sigs.0, bls_sigs.0)),
+				Some(ECDSABLSSignaturePair(ecdsa_sigs.1, bls_sigs.1)),
+			],
+		}
+	}
+
 	#[test]
 	fn should_convert_signed_commitment_to_witness() {
 		// given
-		let signed = signed_commitment();
+		let signed = ecdsa_signed_commitment();
 
 		// when
-		let (witness, signatures) =
-			TestSignedCommitmentWitness::from_signed(signed, |sigs| sigs.to_vec());
+		let (witness, signatures) = TestSignedCommitmentWitness::from_signed::<
+			_,
+			_,
+			Vec<Option<ecdsa_crypto::Signature>>,
+		>(signed, |sigs| sigs.to_vec());
 
 		// then
-		assert_eq!(witness.signatures_merkle_root, signatures);
+		assert_eq!(witness.aggregated_signature, signatures);
+	}
+
+	#[test]
+	fn should_convert_dually_signed_commitment_to_witness() {
+		// given
+		let signed = ecdsa_and_bls_signed_commitment();
+
+		// when
+		let (witness, signatures) = TestBLSSignedCommitmentWitness::from_signed::<
+			_,
+			_,
+			[u8; BLS377::SIGNATURE_SERIALIZED_SIZE],
+		>(signed, |sigs| {
+			//we are going to aggregate the signatures here
+			let mut aggregatedsigs: SignatureAggregatorAssumingPoP<BLS377> =
+				SignatureAggregatorAssumingPoP::new();
+			sigs.iter().filter_map(|sig| {
```

Comment:
not only I assumed the commitment is the aggregated signature, I tought is should be computed only by the prover not each validator. I think @AlistairStewart says that this is the commitment and it should be computed and sign by all validators: 
![commitment-to-public-keys](https://user-images.githubusercontent.com/35698397/217512909-7959924a-9369-4f43-aea7-7152d5e2485d.png)

Reply:
I added the commitment to the set of public keys here: https://github.com/paritytech/substrate/pull/13311/files#diff-90d0377841064e4bcca6f0956bf16bd71e11159f9dd6d5975b91f2b4d901b0a4R200 

It is not clear for me where it should be actually computed and updated.